src/pages/publication/PublicationFormValidationSchema.ts

```diff
@@ -17,6 +17,71 @@ const ErrorMessage = {
   MUST_BE_FUTURE: i18n.t('publication:feedback.date_must_be_in_future'),
 };
 
+const contributorValidationSchema = {
+  correspondingAuthor: Yup.boolean(),
+  email: Yup.string().when('correspondingAuthor', {
+    is: true,
+    then: Yup.string().email(ErrorMessage.INVALID_FORMAT).required(ErrorMessage.REQUIRED),
+  }),
+  sequence: Yup.number().min(0),
+};
+
+const fileValidationSchema = {
+  administrativeAgreement: Yup.boolean(),
+  embargoDate: Yup.date()
+    .nullable()
+    .when('administrativeAgreement', {
+      is: false,
+      then: Yup.date().nullable().min(new Date(), ErrorMessage.MUST_BE_FUTURE).typeError(ErrorMessage.INVALID_FORMAT),
+    }),
+  publisherAuthority: Yup.boolean()
+    .nullable()
+    .when('administrativeAgreement', {
+      is: false,
+      then: Yup.boolean().required(ErrorMessage.REQUIRED),
+    }),
+  license: Yup.object()
+    .nullable()
+    .when('administrativeAgreement', {
+      is: false,
+      then: Yup.object().required(ErrorMessage.REQUIRED),
+    }),
+};
+
+const journalContentValidationSchema = {
+  type: Yup.string().oneOf(Object.values(JournalArticleType)).required(ErrorMessage.REQUIRED),
+  peerReviewed: Yup.boolean().required(ErrorMessage.REQUIRED),
+  articleNumber: Yup.string(),
+  volume: Yup.string(),
+  issue: Yup.string(),
+  pages: Yup.object().shape({
+    begin: Yup.string(),
+    end: Yup.string(),
+  }),
+};
+
+const bookValidationSchema = {
+  type: Yup.string().oneOf(Object.values(BookType)).required(ErrorMessage.REQUIRED),
+  isbn: Yup.string(),
+  numberOfPages: Yup.string(),
+  textBook: Yup.boolean(),
+  peerReviewed: Yup.boolean().required(ErrorMessage.REQUIRED),
+};
+
+const reportValidationSchema = {
+  type: Yup.string().oneOf(Object.values(ReportType)).required(ErrorMessage.REQUIRED),
+  isbn: Yup.string(),
+  numberOfPages: Yup.string(),
+};
+
+const degreeValidationSchema = {
+  type: Yup.string().oneOf(Object.values(DegreeType)).required(ErrorMessage.REQUIRED),
+};
+
+const chapterValidationSchema = {
+  type: Yup.string().length(0),
```

Comment:
hva betyr dette? tom streng?

Reply:
ja, vet ikke om noen subtype for chapter, så sikrer bare at den er tom. egentlig bare JournalArticle som har reell validering atm